src/runtime/TemporalObject.cpp

```diff
@@ -68,228 +68,284 @@ Value TemporalObject::toISOTime(ExecutionState& state, DateObject& d)
     return Value(new ASCIIString(result.data(), result.length()));
 }
 
-std::map<std::string, std::string> TemporalObject::parseValidIso8601String(ExecutionState& state, const Value& str)
+
+std::string TemporalObject::getNNumberFromString(std::string& isoString, const int n)
 {
-    ASSERT(str.isString());
-    size_t pos = str.asString()->toNonGCUTF8StringData().find('[');
-    std::string isoDate = str.asString()->toNonGCUTF8StringData().substr(0, pos);
-    std::string extra;
+    std::string retVal;
+    if (!TemporalObject::is_number(isoString.substr(0, n))) {
+        throw std::exception();
+    }
+    retVal = isoString.substr(0, n);
+    isoString = isoString.substr(n);
```

Comment:
A generic question: this method requires a lot of string  reallocation, which is bad for performance. Would it be possible to use an offset or char pointer?

Reply:
Yes i will use offsets.